### ISSUE TITLE
Return `String` should be nullable, no?

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/web#309]
https://github.com/dart-lang/web/blob/6538aca01c8b3edb00e085d7e349e61f04a746a0/web/lib/src/dom/html.dart#L1281

CC @srujzs

**Comment by srujzs** [comment]
This is similar to https://github.com/dart-lang/web/issues/181. The [IDL](https://github.com/w3c/webref/blob/f2741e4c1aff2765a0313092457af610d2cd0ce7/ed/idl/html.idl#L164) claims the return type is non-nullable, but naturally it would be nice to coerce missing values to `null` instead of a cast failure. I'm not sure if the behavior is similar everywhere a `getter` is declared in the IDL to make that change. The current solution is using `has` to check for existence.

**Comment by rutvik110** [comment]
> I'm not sure if the behavior is similar everywhere a getter is declared in the IDL to make that change. The current solution is using has to check for existence.

From what I observed in the idl specs, the behaviour isn't consistent. Some getters are non-nullable while some are defined nullable. I don't know yet what determines that though. 

But it would be nice to make this nullable as the users won't have to remember to check for the nullable case by using `has`, it would be explicit by the returned value from the getter. I could contribute to this issue if you could share what changes we might be looking towards fixing this issue beside just making this getter nullable.

**Comment by isoos** [comment]
This is a super surprising bug that prevents easy migration. From the API I'd expect it to be returning empty value or throw an Exception, but instead we are getting `null` when the value is missing. If this is an edge case, it may not be covered in tests and will cause a production issue later down the line.

Please fix this, or at minimum put up big warning signs in the documentation.

**Comment by srujzs** [comment]
> From the API I'd expect it to be returning empty value or throw an Exception, but instead we are getting null when the value is missing.

Sorry - for clarification, you should be seeing an exception in Dart when the value is missing. Specifically, you should see a cast from null to `String` exception. Are you seeing something else or are you saying the JS API itself doesn't throw an exception (which is true, it should return `undefined`)?

> Please fix this, or at minimum put up big warning signs in the documentation.

I'm okay adding a warning here for this getter that the code will throw if the value is missing or `null`.

**Comment by isoos** [comment]
> Specifically, you should see a cast from null to String exception.

Yes, and my first intuitive solution was to cast it to `String?` - which did work. I was tempted to leave as-is, however, I did dig deeper until I've found this issue about it, and then started fixing it with the `has` call before using the value getter. But I am sure this is not the right developer experience, and I think the API should be true to the real type of the returned value: `String?`.
